ops/lib/__init__.py

```diff
@@ -0,0 +1,168 @@
+        raise ValueError("invalid library name: {} (chars and digits only)".format(name))
+    if not _libauthor_re.match(author):
+        raise ValueError("invalid library author email: {}".format(author))
+    if not isinstance(api, int):
+        raise ValueError("invalid library API: {} (must be an int)".format(api))
+
+    versions = _libraries.get((name, author), ())
+    for lib in versions:
+        if lib.api == api:
+            return lib.import_module()
+
+    others = ', '.join(str(lib.api) for lib in versions)
+    if others:
+        raise RuntimeError('cannot find "{}" from {} with API {} (have {})'.format(
+            name, author, api, others))
+    else:
+        raise RuntimeError('cannot find library "{}" from {}'.format(
+            name, author))
+
+
+def autoimport():
+    _libraries.clear()
+    for sys_dir in sys.path:
+        if not isinstance(sys_dir, str):
+            continue
+        if sys_dir == "":
+            sys_dir = "."
+        try:
+            top_dirs = os.listdir(sys_dir)
+        except OSError:
+            continue
+        for top_dir in top_dirs:
+            try:
+                lib_dirs = os.listdir(os.path.join(sys_dir, top_dir, 'opslib'))
+            except OSError:
+                continue
+            finder = get_importer(os.path.join(sys_dir, top_dir, 'opslib'))
+            if finder is None or not hasattr(finder, 'find_spec'):
+                continue
+            for lib_dir in lib_dirs:
+                spec = finder.find_spec(lib_dir)
+                if spec is None:
+                    continue
+                lib = _parse_lib(spec)
+                if lib is None:
+                    continue
+
+                versions = _libraries.setdefault((lib.name, lib.author), [])
+                versions.append(lib)
+                versions.sort(reverse=True)
+
+
+def _parse_lib(spec):
+    try:
+        with open(spec.origin) as f:
+            name = author = api = patch = None
+            for line in f:
+                m = _libline_re.match(line)
+                if not m:
+                    continue
+                g = m.groups()
+                key, value = g[0], g[1]
+                if key == "NAME":
+                    name = value
+                    continue
+                if key == "AUTHOR":
+                    author = value
+                    continue
+                if key == "API":
+                    api = value
+                    continue
+                if key == "PATCH":
+                    patch = value
+                    continue
+                if name and author and api and patch:
+                    break
+    except OSError:
+        return None
+
+    if not (name and author and api and patch):
+        return None
+
+    try:
+        # We could easily parse it but it'd still be more work than this,
+        # and given the strict regexp above, this should hopefully be okay.
+        empty = {}
+        name = eval(name, empty)
```

Comment:
For this use case it's better to use `ast.literal_eval`, it's safer and no losing functionality.

Reply:
I'm thinking it's going to be safer, and less surprising, to just use py_compile here.

Reply:
`ast.literal_eval` looks exactly right for this purpose. Thanks for the suggestion @facundobatista.